include/deal.II/lac/precondition.h

```diff
@@ -603,7 +697,42 @@ class PreconditionPSOR : public PreconditionRelaxation<MATRIX>
   /**
    * Declare type for container size.
    */
-  typedef types::global_dof_index size_type;
+  typedef typename MATRIX::size_type size_type;
+
+  /**
+    * Parameters for PreconditionPSOR.
+    */
+  class AdditionalData
+  {
+  public:
+    /**
+     * Constructor. For the parameters' description, see below.
+     *
+     * The permutation vectors are stored as a reference. Therefore, it has to
+     * be assured that the lifetime of the vector exceeds the lifetime of the
+     * preconditioner.
+     *
+     * The relaxation parameter should be larger than zero and smaller than 2
+     * for numerical reasons. It defaults to 1.
+     */
+    AdditionalData (const std::vector<size_type> &permutation,
+                    const std::vector<size_type> &inverse_permutation,
+                    const typename PreconditionRelaxation<MATRIX>::AdditionalData
+                    &parameters = typename PreconditionRelaxation<MATRIX>::AdditionalData());
+
+    /**
+     * Storage for the permutation vector.
+     */
+    const std::vector<size_type> &permutation;
+    /**
+     * Storage for the inverse permutation vector.
+     */
+    const std::vector<size_type> &inverse_permutation;
+    /**
+     * Relaxation parameters
+     */
+    typename PreconditionRelaxation<MATRIX>::AdditionalData parameters;
```

Comment:
Instead of storing this object, why not simply derive this class from `PreconditionRelaxtion<...>::AdditionalData`?